react-validated-input/my-app/src/ValidatedInput.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+import { FaCheck, FaXmark } from 'react-icons/fa6';
+import './ValidatedInput.css';
+
+export function ValidatedInput() {
+  const [password, setPassword] = useState('');
+
+  const specialChars = [
+    '!',
+    '@',
+    '#',
+    '$',
+    '%',
+    '^',
+    '&',
+    '*',
+    '(',
+    ')',
+    ';',
+    '-',
+    '_',
+    '+',
+    '=',
+    '/',
+    '?',
+    '<',
+    '>',
+    '[',
+    ']',
+    '{',
+    '}',
+    '\\',
+    '|',
+  ];
+
+  const errorArr: string[] = [];
+  if (password.length === 0) {
+    errorArr.push('no password');
+  } else if (password.length < 8) {
+    errorArr.push('too short');
+  }
+
+  let hasNoNums = true;
+  let hasNoCapitals = true;
+  let hasNoSpecialChars = true;
+  for (let i = 0; i < password.length; i++) {
+    if (Number(password[i]) || Number(password[i]) === 0) {
+      hasNoNums = false;
+    } else if (specialChars.includes(password[i])) {
+      hasNoSpecialChars = false;
+    } else if (
+      !specialChars.includes(password[i]) &&
+      password[i] === password[i].toUpperCase()
+    ) {
+      hasNoCapitals = false;
+    }
+  }
+  if (hasNoNums) errorArr.push('has no digits');
+  if (hasNoCapitals) errorArr.push('has no capital letters');
+  if (hasNoSpecialChars) errorArr.push('has no special characters');
+
+  let errorMessage = '';
```

Comment:
I think you can simplify this with something like:
`const errorMessage = errorArr.join(', ');`